product/views/item_view.py

```
@@ -95,13 +104,15 @@ class ItemDetailView(APIView):
+    @property
+    def serializer_class(self):
+        return ItemWithImagesSerializer
+
     def get(self, request, item_id):
         try:
             item = Item.objects.get(id=item_id)
         except Item.DoesNotExist:
-            return Response(
-                {"detail": "Item not found."}, status=status.HTTP_404_NOT_FOUND
-            )
+            raise ItemNotFoundException()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)